src/patterns/kinesis-lambda.ts

```diff
@@ -1,18 +1,33 @@
 import { Stream, StreamEncryption } from "aws-cdk-lib/aws-kinesis";
-import type { StreamProps } from "aws-cdk-lib/aws-kinesis";
+import type { IStream, StreamProps } from "aws-cdk-lib/aws-kinesis";
 import { StartingPosition } from "aws-cdk-lib/aws-lambda";
 import { KinesisEventSource } from "aws-cdk-lib/aws-lambda-event-sources";
 import type { KinesisEventSourceProps } from "aws-cdk-lib/aws-lambda-event-sources";
 import type { GuLambdaErrorPercentageMonitoringProps, NoMonitoring } from "../constructs/cloudwatch";
 import { AppIdentity } from "../constructs/core";
 import type { GuMigratingResource, GuStack } from "../constructs/core";
+import { guAssumeRolePolicyStatement } from "../constructs/iam";
 import { GuKinesisStream } from "../constructs/kinesis";
 import type { GuKinesisStreamProps } from "../constructs/kinesis";
 import { GuLambdaFunction } from "../constructs/lambda";
 import type { GuFunctionProps } from "../constructs/lambda";
 import { toAwsErrorHandlingProps } from "../utils/lambda";
 import type { StreamErrorHandlingProps, StreamProcessingProps } from "../utils/lambda";
 
+/**
+ * Used to provide information about an existing cross account Kinesis stream to the [[`ExistingKinesisStream`]] pattern.
+ */
+export interface CrossAccountKinesisStream {
```

Comment:
I am not super keen on passing ARNs around because it tempts people to:

1. hardcode account numbers (which are generally considered private) in the template OR
2. [use CloudFormation parameters](https://github.com/guardian/top-mentions-aggregator/commit/cfd6e12f9c95a52ef718f00a66dc31a71f4085a6#diff-60975a218e4fb7657e3978b4d669f3f64da6c38a13cb81e3f2ce5fb60f6eaf43R20-R27), which is preferable to 1 for privacy reasons, but generally discouraged in `cdk` and also requires manual intervention when deploying

It's a bit beyond the scope of this PR, but I wonder if `@guardian/cdk` should provide helpers for referencing other Guardian accounts. We could, for example, store a map of these in parameter store. So we'd have an interface like:

```typescript 
// In some shared file
type GuardianAwsAccount = "CAPI" | "Mobile"; //etc.

export interface CrossAccountKinesisStream {
  account: GuardianAwsAccount ;
  roleName: string;
  streamName: string;
}
```

That way clients could do something like:

```typescript 
crossAccountKinesisStream: {
  account: "CAPI",
  roleName: "the-role",
  streamName: "the-stream",
},
```

And the ARNs would be built behind the scenes.